services/src/main/java/org/keycloak/services/managers/RealmManager.java

```diff
@@ -422,7 +424,8 @@ private void setupAccountManagement(RealmModel realm) {
             accountClient.setFullScopeAllowed(false);
 
             accountClient.setRootUrl(Constants.AUTH_BASE_URL_PROP);
-            String baseUrl = "/realms/" + realm.getName() + "/account/";
+
+            String baseUrl = UriBuilder.fromPath( "/realms/{realmName}/account/").build(realm.getName()).toString();
```

Comment:
I think that realm name needs decoding in [AdminRoot](https://github.com/keycloak/keycloak/blob/23.0.6/services/src/main/java/org/keycloak/services/resources/admin/AdminRoot.java#L170). Just decode the realm name to transform the `%20` into ` ` (`Encode.decodePath`). With this change I can login OK in account and admin consoles. Please try to add something to test this in your test method (for example create an admin user in the `new realm` and call the `serverinfo` endpoint).

Reply:
@PSanetra I have prepared a branch with the changes that I requested over your commit. It's this [commit](https://github.com/rmartinc/keycloak/commit/ab0544952ed5f4d7e1174b97ab4a73938bc7a428). Can you please rebase this branch with current main and add my commit over it? You can squash them or send both commits, whatever you prefer.

If you are busy I can also close this PR and send a new one with my branch. Your commit will be there too.

Thanks!

Reply:
@PSanetra I'm preparing the PR with both changes, I'll close this as soon as I have the other PR ready saying it's superseded by the new one. Thanks!